src/Microsoft.OData.Edm/Schema/EdmTargetPath.cs

```diff
@@ -0,0 +1,65 @@
+//---------------------------------------------------------------------
+// <copyright file="EdmTargetPath.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.OData.Edm
+{
+    /// <inheritdoc/>
+    public class EdmTargetPath : IEdmTargetPath
+    {
+        private IEnumerable<IEdmElement> segments;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EdmTargetPath"/> class.
+        /// </summary>
+        /// <param name="segments">Target path segments.</param>
+        public EdmTargetPath(IEnumerable<IEdmElement> segments)
+        {
+            EdmUtil.CheckArgumentNull(segments, nameof(segments));
+
+            ValidateSegments(segments);
+
+            this.segments = segments;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EdmTargetPath"/> class.
+        /// </summary>
+        /// <param name="segments">Target path segments.</param>
+        public EdmTargetPath(params IEdmElement[] segments)
+            :this(segments.AsEnumerable())
+        {
+        }
+
+        /// <inheritdoc/>
+        public IEnumerable<IEdmElement> Segments
+        {
+            get { return this.segments; }
+        }
+
+        private static void ValidateSegments(IEnumerable<IEdmElement> segments)
+        {
+            // Validate that the first element is Container.
+            IEdmElement firstElement = segments.FirstOrDefault();
+
+            if (firstElement != null && !(firstElement is IEdmEntityContainer))
+            {
+                throw new InvalidOperationException(Strings.TargetPath_FirstSegmentMustBeIEdmEntityContainer);
+            }
+
+            // Validate the second element is a Container element.
+            IEdmElement secondElement = segments.ElementAt(1);

```

Comment:
>ElementAt(1) [](http://example.com/codeflow?start=49&length=12)

Shall we verify the segments contain more than or equal 2 elements, so, we can safely call 'ElementAt(1)`